client/rest-high-level/src/test/java/org/elasticsearch/client/CrudIT.java

```diff
@@ -265,4 +340,27 @@ public void testIndex() throws IOException {
                          "version conflict, document already exists (current version [1])]", exception.getMessage());
         }
     }
+
+    /**
+     * @return A sample index request
+     * @throws IOException If problem when generating the content
+     */
+    private IndexRequest createSampleIndexRequest() throws IOException {
```

Comment:
> I also wonder if it's ok to always rely on auto-generated ids. maybe we should provide our ids instead.

Why not? 

note that sometimes I'm doing:

```java
createSampleIndexRequest().id(randomAsciiOfLength(15);
```

Where I basically force the id. So I'd say that we have both covered.